src/plugins/typescript/visitors.ts

```diff
@@ -0,0 +1,142 @@
+import { Comment } from "typedoc/dist/lib/models/comments/comment";
+import { DefaultValueContainer } from "typedoc/dist/lib/models/reflections/abstract";
+import {
+    ITsClass,
+    ITsInterface,
+    ITsMethod,
+    ITsMethodSignature,
+    ITsParameter,
+    ITsProperty,
+    Kind,
+} from "../../client/typescript";
+import { ICompiler } from "../plugin";
+import { resolveSignature, resolveTypeString } from "./typestring";
+
+export type Renderer = ICompiler["renderBlock"];
+
+export function visitorExportedClass(def: ContainerReflection, renderer: Renderer): ITsClass {
+    return {
+        ...visitorExportedInterface(def, renderer)!,
+        kind: Kind.Class,
+    };
+}
+
+export function visitorExportedInterface(def: ContainerReflection, renderer: Renderer): ITsInterface {
+    return {
+        documentation: renderComment(def.comment, renderer),
+        fileName: getFileName(def),
+        kind: Kind.Interface,
+        methods: def
+            .getChildrenByKind(ReflectionKind.Method)
+            .filter(child => !isInternal(child, true))
+            .map(method => visitorExportedMethod(method, renderer)),
+        name: def.name,
+        properties: def
+            .getChildrenByKind(ReflectionKind.Property)
+            .filter(child => !isInternal(child, true))
+            .map(prop => visitorExportedProperty(prop, renderer)),
+    };
+}
+
+function visitorExportedProperty(def: DeclarationReflection, renderer: Renderer): ITsProperty {
+    return {
+        defaultValue: getDefaultValue(def),
+        documentation: renderComment(def.comment, renderer),
+        fileName: getFileName(def),
+        kind: Kind.Property,
+        name: def.name,
+        type: resolveTypeString(def.type),
+    };
+}
+
+function visitorExportedMethod(def: DeclarationReflection, renderer: Renderer): ITsMethod {
+    return {
+        fileName: getFileName(def),
+        kind: Kind.Method,
+        name: def.name,
+        signatures: def.signatures.map(sig => visitorSignature(sig, renderer)),
+    };
+}
+
+function visitorSignature(sig: SignatureReflection, renderer: Renderer): ITsMethodSignature {
+    return {
+        documentation: renderComment(sig.comment, renderer),
+        kind: Kind.Signature,
+        parameters: (sig.parameters || []).map(param => visitorParameter(param, renderer)),
+        returnType: resolveTypeString(sig.type),
+        type: resolveSignature(sig),
+    };
+}
+
+function visitorParameter(param: ParameterReflection, renderer: Renderer): ITsParameter {
+    return {
+        defaultValue: getDefaultValue(param),
+        documentation: renderComment(param.comment, renderer),
+        fileName: getFileName(param.parent),
+        flags: param.flags || {},
+        kind: Kind.Parameter,
+        name: param.name,
+        type: resolveTypeString(param.type),
+    };
+}
+
+/**
+ * Converts a typedoc comment object to a rendered `IBlock`.
+ */
+function renderComment(comment: Comment, renderer: Renderer) {
```

Comment:
return typedef?

Reply:
not necessary here as it's straight out of `renderer`. also adding it would require another import.